### PR TITLE
Refactors calculating shuttle bounds into a dedicated proc

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

See title.
Also adds the ability to calculate the information during runtime by iterating through the shuttle areas; which essentially means that you now have the ability to shrink/expand shuttles. A front end for doing this has not been implemented in this PR, however this means that in the future doing so would be as simple as adding the new area into the shuttle_areas list on the docking port and then calling `calculate_docking_port_information` on the docking port.

## Why It's Good For The Game

Adds a backend for expanding or shrinking a shuttle.
Moves logic for calculating a mobile port's w, h, dx, dy to a dedicated proc

## Changelog
No player facing stuff, yet.

### CODE REVIEW

File: code/modules/shuttle/shuttle.dm [hunk 1]
```diff
@@ -431,6 +431,69 @@
...
+ *
+ * Arguments:
+ * * loading_from - The template that the shuttle was loaded from, if not given we iterate shuttle_areas to calculate information instead
+ */
+/obj/docking_port/mobile/proc/calculate_docking_port_information(datum/map_template/shuttle/loading_from)
+	var/port_x_offset = loading_from?.port_x_offset
+	var/port_y_offset = loading_from?.port_y_offset
+	var/width = loading_from?.width
+	var/height = loading_from?.height
+	if(!istype(loading_from))
```
Comment (JohnFulpWillard) [inline code/modules/shuttle/shuttle.dm:448]: When would this not be a shuttle map template and why do we want to support that being the case?

Comment (ZephyrTFA) [reply code/modules/shuttle/shuttle.dm:448]: This is just checking for null, I'm not sure why I didn't just use ! and I also don't remember

Comment (ZephyrTFA) [reply code/modules/shuttle/shuttle.dm:448]:
```suggestion
	if(!loading_from)
```

File: code/modules/shuttle/shuttle.dm [hunk 2]
```diff
@@ -431,6 +431,69 @@
...
+ *
+ * Arguments:
+ * * loading_from - The template that the shuttle was loaded from, if not given we iterate shuttle_areas to calculate information instead
+ */
+/obj/docking_port/mobile/proc/calculate_docking_port_information(datum/map_template/shuttle/loading_from)
+	var/port_x_offset = loading_from?.port_x_offset
+	var/port_y_offset = loading_from?.port_y_offset
+	var/width = loading_from?.width
+	var/height = loading_from?.height
+	if(!loading_from)
```
Comment (JohnFulpWillard) [inline code/modules/shuttle/shuttle.dm:448]: This just confuses me even more, why do we want to support them NOT having a template at all?

**Pull request merged**